x-pack/plugins/cases/common/api/cases/case.ts

```diff
@@ -258,11 +258,11 @@ export const CasesFindRequestRt = rt.exact(
     /**
      * The page of objects to return
      */
-    page: NumberFromString,
+    page: rt.union([rt.number, NumberFromString]),
```

Comment:
I wanted to add a comment and forgot 😛 

This is because of the tests.

Check the test `'Invalid total items results in error'` in `x-pack/plugins/cases/server/client/cases/find.test.ts` for example.

We need to define a `page` and `perPage` in the `createCasesClientMockFindRequest` which expects params of type `CasesFindRequest`.

The problem is that `type CasesFindRequest = rt.TypeOf<typeof CasesFindRequestRt>;` which converts 
`page: NumberFromString` and `perPage: NumberFromString` to `number`.

So a number was expected for these parameters.

There are then two scenarios:
1. If I passed a number, Typescript didn't complain but the test failed on decode because `Error: Invalid value "some-number" supplied to \"page\" or \"perPage\"`(we are decoding with `CasesFindRequestRt` which expects a `string`)
2. If I passed a string the test passed but we got a Typescript error.

**I wanted to avoid the typescript ignore** so I allowed these params to be numbered, which is harmless because we expect `NumberFromString` anyway.

I was surprised we didn't have this problem in other places but there are not many tests for `page` and `perPage`. However, turns out `FindCommentsQueryParamsRt` does the same.

Reply:
Thanks for the detailed answer! Makes sense now.